### ISSUE TITLE
use variadic argument to determine whether the new perf API is in use

### ISSUE THREAD

**Issue by ThinkOpenly** [open-power-sdk/curt#34]
Currently, determining which perf API to use is left to the user, with the `--api` parameter, with the default being the older API which works everywhere, but which does not report process-specific information.

Use variadic arguments instead, like:
```
def irq__irq_handler_exit(event_name, context, common_cpu, common_secs, common_nsecs, common_pid, common_comm, common_callchain, irq, ret, *args):
```
with `perf_sample_dict` existing as the first tuple of `args`, or not existing.  The pid can then either be determined, or set to `'unknown'` based on that.

**Comment by ThinkOpenly** [comment]
Unfortunately, I don't think this will work because of this code in tools/perf/util/scripting-engines/trace-event-python.c:
```
	if (get_argument_count(handler) == (int) n + 1) {
		all_entries_dict = get_perf_sample_dict(sample, evsel, al,
			callchain);
		PyTuple_SetItem(t, n++,	all_entries_dict);
	}
```
`perf` checks the number of arguments before sending `perf_sample_dict` to the event handler.  I'm ot sure why it bothers, but unfortunately, that prevents a variadic argument from optionally receiving the extra argument.

**Comment by ThinkOpenly** [comment]
this is now fixed using a try-and-fail method by commit de370eb633c93ca39b851eb764d407fd39558966
